source/tutorials/kubernetes.rst

```diff
@@ -3,170 +3,340 @@
-Learn about the benefits of having an up-to-date system for cloud
-orchestration on the :ref:`swupd-guide` page.
+instructions on installing |CL| on a bare metal system, follow the :ref:`bare
+metal installation tutorial<bare-metal-install-desktop>`.
 
-Before you install any new packages, update |CL| with the following command:
+Before you continue, update your |CL| installation with the following command:
 
 .. code-block:: bash
 
    sudo swupd update
 
-Install Kubernetes and CRI runtimes
-***********************************
+Learn about the benefits of having an up-to-date system for cloud
+orchestration on the :ref:`swupd-guide` page.
+
+
+Install Kubernetes
+******************
 
 Kubernetes, a set of supported :abbr:`CRI (Container Runtime Interface)`
-runtimes, and networking plugins, are included in the `cloud-native-basic`_
-bundle.
+runtimes and :abbr:`CNI (Container Network Interface)` are included in the
+`cloud-native-basic`_ bundle.
 
-To install this framework, enter the following command:
+#. Install the cloud-native-basic bundle to get these components:
 
 .. code-block:: bash
 
    sudo swupd bundle-add cloud-native-basic
 
-.. note::
 
-   For more on networking plugins, see `Install pod network add-on`_.
+Configure host system
+=====================
 
-Configure Kubernetes
-********************
+This tutorial uses the basic default Kubernetes configuration for to get
+started. You can customize your Kubernetes configuration according to your
+specific deployment and security needs.
 
-This tutorial uses the basic default Kubernetes configuration for simplicity.
-You must define your Kubernetes configuration according to your specific
-deployment and your security needs.
+The Kubernetes administration tool, :command:`kubeadm`, performs some
+"`preflight checks`_" when initializing and starting a cluster. The steps
+below are necessary to ensure those `preflight checks`_ pass successfully.
 
-#. Enable IP forwarding to avoid kubeadm `preflight check`_ errors:
 
-   Create (or edit if it exists) the file :file:`/etc/sysctl.d/60-k8s.conf`
-   and include the following line:
+#. Enable IP forwarding:
 
-   .. code-block:: bash
+   - Create the file :file:`/etc/sysctl.d/60-k8s.conf` to set the
+     :command:`net.ipv4.ip_forward` parameter
+
+     .. code-block:: bash
+
+        sudo mkdir -p /etc/sysctl.d/
+
+        sudo tee /etc/sysctl.d/99-kubernetes-cri.conf > /dev/null <<EOF
+        net.bridge.bridge-nf-call-iptables  = 1
+        net.ipv4.ip_forward                 = 1
+        net.bridge.bridge-nf-call-ip6tables = 1
+        EOF
+
+   - Apply the change:
+
+     .. code-block:: bash
+
+        sudo sysctl --system
 
-      net.ipv4.ip_forward = 1
 
-   Apply the change:
+#. Disable swap:
 
    .. code-block:: bash
 
-      sudo systemctl restart systemd-sysctl
+      sudo systemctl mask $(sed -n -e 's#^/dev/\([0-9a-z]*\).*#dev-\1.swap#p' /proc/swaps) 2>/dev/null
```

Comment:
I would still have a small section header or bolded heading that says "Temporary" and "Permanent"

Reply:
I'm opinionated that the temporary step was noise in the documentation and not useful for most people who are going to be using a system for a k8s node. I used the temporary steps and then at next reboot it failed citing swap was enabled so it was not useful to me.

If you think there is going to be a population that wants to do this temporarily, then I rather keep it as a single step + note to not run the mask command.

Reply:
I think then that putting the `sudo swapoff -a` above the `mask` step with a small header that says `Disable Temporarily` and then right below that have `Disable Permanently` as a small header above the `mask` step.